### ISSUE TITLE
null ref in DefineSequencePoints

### ISSUE THREAD

**Issue by SimonCropp** [jbevain/cecil#302]
When i run this https://github.com/Fody/Visualize/tree/newcecil

i get a null ref deep in inside cecil 

this is done using https://github.com/jbevain/cecil/commit/68bcb750b898f4882a5af44299bb322aaa531f93

`sequence_points` is null

See below

I can debug into this, so let me know what other info u need

```
>	Mono.Cecil.Pdb.dll!Mono.Cecil.Pdb.PdbWriter.DefineSequencePoints(Mono.Collections.Generic.Collection<Mono.Cecil.Cil.SequencePoint> sequence_points) Line 84	C#
 	Mono.Cecil.Pdb.dll!Mono.Cecil.Pdb.PdbWriter.Write(Mono.Cecil.Cil.MethodDebugInformation info) Line 50	C#
 	Mono.Cecil.dll!Mono.Cecil.Cil.CodeWriter.WriteUnresolvedMethodBody(Mono.Cecil.MethodDefinition method) Line 95	C#
 	Mono.Cecil.dll!Mono.Cecil.Cil.CodeWriter.WriteMethodBody(Mono.Cecil.MethodDefinition method) Line 51	C#
 	Mono.Cecil.dll!Mono.Cecil.MetadataBuilder.AddMethod(Mono.Cecil.MethodDefinition method) Line 1629	C#
 	Mono.Cecil.dll!Mono.Cecil.MetadataBuilder.AddMethods(Mono.Cecil.TypeDefinition type) Line 1621	C#
 	Mono.Cecil.dll!Mono.Cecil.MetadataBuilder.AddType(Mono.Cecil.TypeDefinition type) Line 1443	C#
 	Mono.Cecil.dll!Mono.Cecil.MetadataBuilder.AddTypeDefs() Line 1414	C#
 	Mono.Cecil.dll!Mono.Cecil.MetadataBuilder.BuildTypes() Line 1269	C#
 	Mono.Cecil.dll!Mono.Cecil.MetadataBuilder.BuildModule() Line 1038	C#
 	Mono.Cecil.dll!Mono.Cecil.ModuleWriter.BuildMetadata.AnonymousMethod__1_0(Mono.Cecil.MetadataBuilder builder, Mono.Cecil.MetadataReader _) Line 134	C#
 	Mono.Cecil.dll!Mono.Cecil.ModuleDefinition.Read<System.__Canon, System.__Canon>(System.__Canon item, System.Func<System.__Canon, Mono.Cecil.MetadataReader, System.__Canon> read) Line 939	C#
 	Mono.Cecil.dll!Mono.Cecil.ModuleWriter.WriteModuleTo(Mono.Cecil.ModuleDefinition module, Mono.Disposable<System.IO.Stream> stream, Mono.Cecil.WriterParameters parameters) Line 108	C#
 	Mono.Cecil.dll!Mono.Cecil.ModuleDefinition.Write(string fileName, Mono.Cecil.WriterParameters parameters) Line 1148	C#
```

![image](https://cloud.githubusercontent.com/assets/122666/19755348/f219300c-9c60-11e6-8abd-4e6090e089ca.png)

**Comment by SimonCropp** [comment]
By process of elimination the offending assembly i am trying to round trip has this call

```
    public class Bar
    {
        public IEnumerator Foo()
        {
            yield break;
        }
    }
```

changing it to 
```
    public class Bar
    {
        public IEnumerator Foo()
        {
            return null;
        }
    }
```
and i do not get the null ref

**Comment by SimonCropp** [comment]
so it seems the null ref is cause by this escape clause 

https://github.com/jbevain/cecil/blob/master/symbols/pdb/Mono.Cecil.Pdb/PdbReader.cs#L186


![image](https://cloud.githubusercontent.com/assets/122666/19756083/e780d532-9c65-11e6-97c2-0f5fd75920a3.png)

**Comment by jbevain** [comment]
Thanks for the super detailed bug report @SimonCropp! <3

**Comment by jbevain** [comment]
Fixed in 155beb470f5aa267b1b42a793d936567fa46cb61.
